controllers/subscription_controller.go

```diff
@@ -118,17 +134,17 @@ func (r *SubscriptionReconciler) ensureSubscriptions(logger logr.Logger, olmPkgR
 	return combinedErr
 }
 
-func (r *SubscriptionReconciler) setOperatorCondition(logger logr.Logger, olmPkgRecords []*OlmPkgRecord) error {
+func (r *SubscriptionReconciler) setOperatorCondition(logger logr.Logger, csvNames []string) error {
 	ocdList := &opv2.OperatorConditionList{}
 	err := r.Client.List(context.TODO(), ocdList, client.InNamespace(r.OperatorNamespace))
 	if err != nil {
 		logger.Error(err, "failed to list OperatorConditions")
 		return err
 	}
 
-	condMap := make(map[string]struct{}, len(olmPkgRecords))
-	for _, olmPkgRecord := range olmPkgRecords {
-		condMap[olmPkgRecord.Csv] = struct{}{}
+	condMap := make(map[string]struct{}, len(csvNames))
+	for _, csvName := range csvNames {
+		condMap[csvName] = struct{}{}
```

Comment:
If we need a map/set for this controller, there is no point in reading the config map CSV names into a slice then converting into a map/set. You should be reading directly into a map/set

Reply:
Done